mypy/checkexpr.py

```diff
@@ -347,6 +347,24 @@ def check_call(self, callee: Type, args: List[Expression],
                     callee.type_object().name(), type.abstract_attributes,
                     context)
 
+            if isinstance(callee, TypedDictGetFunction):
+                if 1 <= len(args) <= 2 and isinstance(args[0], (StrExpr, UnicodeExpr)):
+                    return_type = self.get_typeddict_index_type(callee.typed_dict, args[0])
+                    arg_types = callee.arg_types
+                    if len(args) == 1:
+                        return_type = UnionType.make_union([
+                            return_type, NoneTyp()])
+                    elif isinstance(return_type, TypedDictType) and len(callee.arg_types) == 2:
+                        # Explicitly set the type of the default parameter to
+                        # Union[typing.Mapping, <return type>] in cases where the return value
+                        # is a typed dict.  This special case allows for chaining of `get` methods
+                        # when accessing elements deep within nested dictionaries in a safe and
+                        # concise way without having to set up exception handlers.
+                        arg_types = [callee.arg_types[0],
+                                     UnionType.make_union([return_type,
+                                                           self.named_type('typing.Mapping')])]
```

Comment:
It would better to use the type such as `Mapping[str, Any]`. Not having type arguments for `Mapping` may cause trouble in some cases (e.g. index errors). This still wouldn't be quite safe, but let's deal with that as a separate issue.